internals/overlord/servstate/manager.go

```diff
@@ -267,22 +163,18 @@ const (
 // Services returns the list of configured services and their status, sorted
 // by service name. Filter by the specified service names if provided.
 func (m *ServiceManager) Services(names []string) ([]*ServiceInfo, error) {
-	mgrPlan, err := m.Plan()
-	if err != nil {
-		return nil, err
-	}
-
 	m.servicesLock.Lock()
 	defer m.servicesLock.Unlock()
 
+	planSnapshot := m.plan()
```

Comment:
I argued about this with myself and decided that since the ```servicesLock``` is likely more contested, and the ```planLock``` practically not really ever contested (due to time it takes to copy a pointer), that although never wrong, I convinced myself the service manager services and plan would be closer in sync this way around when the API is used concurrently between service and plan calls. I could be completely wrong.

Reply:
Fixed. Resolving.